lib/iiif_manifest/v3/manifest_builder/canvas_builder.rb

```diff
@@ -54,6 +56,15 @@ def apply_record_properties
           canvas.label = ManifestBuilder.language_map(record.to_s) if record.to_s.present?
           annotation_page['id'] = "#{path}/annotation_page/#{annotation_page.index}"
           canvas.items = [annotation_page]
+          canvas.thumbnail = [thumbnail] if iiif_endpoint
+        end
+
+        def thumbnail
+          if display_image
+            thumbnail_builder_factory.new(display_image).apply(canvas)
+          elsif display_content
+            thumbnail_builder_factory.new(display_content.first).apply(canvas)
```

Comment:
There's a mismatch of logic on `elsif display_content` and the later `display_content.first`.

My strong preference is to have the following (which would likely change the implementation details of display_content).

```ruby
        elsif display_content
             thumbnail_builder_factory.new(display_content).apply(canvas)
```

The reason is that I'm unclear what the return value is for display content.  When I see `if display_content` I assume it will either be an object or `nil`.  And then when I read `display_content.first` I think we must have an array.  Abut `if []` is true, but `[].first` is `nil`.